components/TransactionForm.js

```diff
@@ -0,0 +1,351 @@
+import { categories } from "@/lib/categories";
+import { useState } from "react";
+import CurrencyInput from "react-currency-input-field";
+import styled from "styled-components";
+
+export default function TransactionForm({ onAdd }) {
+  const today = new Date().toISOString().split("T")[0];
+
+  const [selectedType, setSelectedType] = useState("");
+  const [amount, setAmount] = useState("");
+  const [typeError, setTypeError] = useState(false);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    if (data.amount) {
+      data.amount = parseFloat(
+        data.amount.replace(/\./g, "").replace(",", ".")
+      );
+    }
+
+    if (!selectedType) {
+      setTypeError(true);
+      return;
+    }
+
+    onAdd(data);
+
+    event.target.reset();
+    setAmount("");
+    setSelectedType("");
+    setSelectedType(false);
+  }
+
+  return (
+    <>
+      <StyledForm onSubmit={handleSubmit}>
+        <StyledFieldset>
+          <StyledLegend>Add a new transaction</StyledLegend>
+          <FormRow>
+            <StyledNameLabel htmlFor="name">Name</StyledNameLabel>
+            <StyledNameInput
+              type="text"
+              id="name"
+              name="name"
+              maxLength={40}
+              required
+            ></StyledNameInput>
+          </FormRow>
+          <FormRow>
+            <StyledAmountLabel htmlFor="amount">Amount</StyledAmountLabel>
+            <StyledCurrencyInput
+              id="amount"
+              name="amount"
+              maxLength="12"
+              intlConfig={{ locale: "de-DE", currency: "EUR" }}
+              value={amount}
+              onValueChange={(value) => {
+                if (value > 0) {
+                  setAmount(value);
+                } else {
+                  setAmount("");
+                }
+              }}
```

Comment:
https://www.npmjs.com/package/react-currency-input-field#props please use the "allowNegativeValue" prop instead.